packages/components/src/radio-control/index.js

```diff
@@ -20,7 +20,7 @@ export default function RadioControl( { label, className, selected, help, onChan
 	const onChangeValue = ( event ) => onChange( event.target.value );
 
 	return ! isEmpty( options ) && (
-		<BaseControl label={ label } id={ id } help={ help } className={ classnames( className, 'components-radio-control' ) }>
+		<BaseControl as="fieldset" label={ label } id={ id } help={ help } className={ classnames( className, 'components-radio-control' ) }>
```

Comment:
I don't really understand why the RadioControl has to be a fieldset while all other controls are actual controls "fields + labels". I guess the answer is related to a11y. but it makes me wonder if the issue is more on the consumer side and not on the reusable UI component.

Mind explaining the reasoning?

Reply:
> I don't really understand why the RadioControl has to be a fieldset while all other controls are actual controls "fields + labels". I guess the answer is related to a11y. but it makes me wonder if the issue is more on the consumer side and not on the reusable UI component.

Fieldset in general groups together controls. In the case of radio buttons, they directly relate, like described originally: Question plus a set of answers of which only one can be selected.

But you can also use fieldsets to group together pairs of label/inputs that are related to each other. For example, you could have two fieldsets, one for a billing address, the other for a shipping address. Both would have labels+input fields such as last name, first name, street address, zip code, city, etc., but one would have a legend of Shipping Address, and the other of Billing Address. Both are valid use cases for using fieldset/legend. In the case of radio buttons, strictly spec compliant speaking, this is required, in the case of sets of similar groups of fields, it is strongly recommended to do it this way.

Reply:
Thanks for the explanation, I guess my reasoning here is that I don't see the RadioControl (multiple answers to one question) as multiple controls, It is the case HTML wise though, thus the confusion.

I still wonder if it should use BaseControl though or should be a separate component.

Reply:
Maybe it would be a better idea to implement it as 2 distinct components:
- `BaseControl`
- `ControlGroup` or something like that

I guess the new component would be an adjusted clone of `BaseControl`. The benefit is that you could reuse this group in other places and it wouldn't create too much confusion when wrapping multiple controls.

Reply:
@gziolo + @youknowriad Thank you for your input!

@gziolo Your suggestion of creating a separate component, but extending/cloning `BaseControl` was something I had thought of as well 😊 .

Except, the name I had was `FieldSet` rather than `ControlGroup` (I think I like `ControlGroup` better).

In addition to `RadioControl` having multiple options, the same thing would apply to a collection of checkboxes. Strangely enough, it doesn't look like we have a multi-checkbox component - or at least, not one that works in the same manner as `RadioControl`

I can certainly make the update of creating a separate component if it makes grouping vs ungrouping controls clearer.

For this PR, I figure I started with something relatively simple, but flexible.